### PR TITLE
Add user avatar to admin site

### LINKED ISSUE #2829: Implement Avatars support
Allow to add a picture for a user.

Display it in admin, next to the name.

Provide Liquid filters to easily create one in templates.

Possible to use it in when a User picker will be started.

Allow multiple implementations.
A first implementation could be to use Gravatar, based on the email.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #2829

### CODE REVIEW

**Comment by agriffard** [comment]
Hello @hishamco 
When you change a .js or .scss file, you need to run the `gulp build` task (after executing a `npm install` at the root of the repo) in order to bundle the change in `TheAdmin.js`.
It might generate more files than needed but yon can discard them.

**Comment by hishamco** [comment]
Ya I know that after I worked on rtl admin theme

Thanks

**Comment by hishamco** [comment]
This is how it looks like:

Avatar

**Comment by hishamco** [comment]
@agriffard seems the Travis failed for something related to GraphQL, so please review the PR

**Comment by hishamco** [comment]
Few comments (may be for future):

- Add Avatar property to User entity
- Use `MD5` at the server (no client-side scripts needed)
- Cache the avatar while it's not frequently changed
- Use TagHelper like what I did here before in [My.AspNetCore.Mvc.TagHelpers](https://github.com/hishamco/My.AspNetCore.Mvc.TagHelpers) repo

**Comment by agriffard** [comment]
Travis error:
```
[xUnit.net 00:00:14.25]     OrchardCore.Tests.Apis.GraphQL.Queries.RecentBlogPostsQueryTests.ShouldListBlogPostWhenCallingAQuery [FAIL]
Failed   OrchardCore.Tests.Apis.GraphQL.Queries.RecentBlogPostsQueryTests.ShouldListBlogPostWhenCallingAQuery
Error Message:
 Assert.Equal() Failure
Expected: 2
Actual:   1
Stack Trace:
   at OrchardCore.Tests.Apis.GraphQL.Queries.RecentBlogPostsQueryTests.ShouldListBlogPostWhenCallingAQuery() in /home/travis/build/OrchardCMS/OrchardCore/test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs:line 45
```
The other PRs don't seem to have this error. We will see next time the PR is changed if it persists.

**Comment by agriffard** [comment]
It could also be interesting to have a Liquid filter to generate a Gravatar. See #2829

**Comment by hishamco** [comment]
It could be, but while `TheAdmin` theme is using Razor, it's better to use `TagHelper` instead

**Comment by hishamco** [comment]
@agriffard where should the liquid filter located? What about my earlier comments?

**Comment by agriffard** [comment]
There could be a Gravatar module with a Liquid Filter inside (not for your purpose but for another usage in a Liquid Editor) and may be other things like a display editor for a TextField that allows to render the email as its corresponding gravatar.

Can you please fix the conflict by regenerating the script.

**Comment by hishamco** [comment]
> Can you please fix the conflict by regenerating the script.

Sure

**Comment by hishamco** [comment]
@agriffard I added a gravatar module, now how can I add the tag helper to the admin site? Shall I reference the module? How can I show the avatar if the module is enabled? May be there's a standard for you guys which I don't know

thanks

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml [hunk 1]
```diff
@@ -32,12 +38,17 @@
                     @await RenderSectionAsync("NavbarTop", required: false)
                 </div>
                 <ul class="nav navbar-nav">
-                    <li class="nav-item">
-                        <a class="nav-link"><span><i class="fa fa-user" aria-hidden="true"></i></span><span> @User.Identity.Name</span></a>
+                    <li class="nav-item dropdown">
```
Comment (jptissot) [inline src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml]: This should probably a shape that get's modified if the Gravatar module is enabled.

Comment (hishamco) [reply src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml]: This is not related to gravatari. IMHO is suited coz it injected and set the image and it's properties

**Comment by hishamco** [comment]
@agriffard what's the action needed here, before resolving the conflict?

**Comment by hishamco** [comment]
@agriffard any updates on this?

**Comment by agriffard** [comment]
@hishamco please fix conflicts, first.

**Comment by agriffard** [comment]
Could you merge dev and display a rounded avatar for the user in the navbar?
Does it work well as a dependency in the admin?
If it was an AdminSettings to enable, that would be great.

File: src/OrchardCore.Themes/TheAdmin/wwwroot/Styles/TheAdmin.css [hunk 2]
```diff
@@ -9178,156 +9331,3 @@ select[multiple] {
     .persona .persona-content.small, .persona .persona-content.hint {
```
Comment (Skrypt) [inline src/OrchardCore.Themes/TheAdmin/wwwroot/Styles/TheAdmin.css:9331]: Styles should not be added in the .css files directly.

Comment (hishamco) [reply src/OrchardCore.Themes/TheAdmin/wwwroot/Styles/TheAdmin.css:9331]: This is too old, I already schedule my time to revise the old PRs

Thanks again

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/wwwroot/Styles/TheAdmin.css:9331]: I agree. I kept the PR opened because there are things that can be reused.

**Comment by agriffard** [comment]
@hishamco Do you plan to make a new PR?

**Comment by hishamco** [comment]
> @hishamco Do you plan to make a new PR?

@agriffard I moved this to Orchard Core Contribute, but if there's an intend to accept this to be a part of OC, I will start a new PR

Thanks

**Comment by agriffard** [comment]
The goal was to have it in the admin for the current user, so yes, it would be great to have it in OC as a feature.

**Comment by hishamco** [comment]
I will start a new PR then

**Pull request closed without merging**